### PR TITLE
Fix contract details

### PR DESCRIPTION
**Pull request by zaelgohary**
### Description

Error happened because getGPUInfo was being called with an undefined nodeId. 

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/881
- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/741

### Checklist

- [ ] Tests included
- [ ] Build pass
- [ ] Documentation
- [ ] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

**Review by AhmedHanafy725 — COMMENTED** [review]
in case the full vm doesn't have a gpu and tries to load the info from the deployment list, it keeps loading forever.

image

File: packages/playground/src/weblets/tf_contracts_list.vue [hunk 1]
```diff
@@ -162,17 +162,31 @@ const loadingContractId = ref<number>();
...
+        graphqlURL: grid!.clientOptions!.graphqlURL!,
+      });
+    }
+
+    if (value.type === "rent") {
+      deployment = await grid!.contracts.getRentContractByContractId({
+        contractId,
+        graphqlURL: grid!.clientOptions!.graphqlURL!,
+      });
+    }
```
Comment (AhmedHanafy725) [inline packages/playground/src/weblets/tf_contracts_list.vue:187]: why do we do these calls we already have the info we need to show?

File: packages/playground/src/components/deployment_data_dialog.vue [hunk 2]
```diff
@@ -201,27 +201,34 @@ async function getGrafanaUrl() {
...
+    }
   }
 
   if (gpuInfo.value == "") {
     showGpuCard.value = false;
   }
   loadingCard.value = false;
 }
-getGPUInfo();
+contract.value.metadata?.includes("fullvm") ? getGPUInfo() : undefined;
```
Comment (0oM4R) [inline packages/playground/src/components/deployment_data_dialog.vue:231]: __Suggestion__: this line is functional and will work, but i believe it's not the most clear or common way to structure this kind of logic. there are more readable alternatives that can improve code readability and maintainability
for example: 
`if(contract.value.metadata?.includes("fullvm"))  getGPUInfo();`

File: packages/grid_client/src/clients/tf-grid/contracts.ts [hunk 3]
```diff
@@ -62,9 +65,13 @@ class TFContracts extends Contracts {
                 }
                 rentContracts(where: {twinID_eq: ${options.twinId}, state_in: ${state}}, limit: $rentContractsCount) {
                   contractID
-                  state
                   createdAt
+                  gridVersion
+                  id
```
Comment (AhmedHanafy725) [inline packages/grid_client/src/clients/tf-grid/contracts.ts:70]: `gridVersion` and `id` are not useful to the user, let's remove them

File: packages/grid_client/src/clients/tf-grid/contracts.ts [hunk 4]
```diff
@@ -48,10 +47,14 @@ class TFContracts extends Contracts {
       const rentContractsCount = await gqlClient.getItemTotalCount("rentContracts", opts);
       const body = `query getContracts($nameContractsCount: Int!, $nodeContractsCount: Int!, $rentContractsCount: Int!){
                 nameContracts(where: {twinID_eq: ${options.twinId}, state_in: ${state}}, limit: $nameContractsCount) {
+                  id
                   contractID
-                  state
-                  name
                   createdAt
+                  gridVersion
```
Comment (AhmedHanafy725) [inline packages/grid_client/src/clients/tf-grid/contracts.ts:53]: here too let's remove `gridVersion` and `id`

**Review by AhmedHanafy725 — CHANGES_REQUESTED** [review]

**Comment by AhmedHanafy725** [comment]
when trying to get the contract's details of a full vm from the contract list, I got this error
image

**Review by AhmedHanafy725 — APPROVED** [review]

**Pull request merged**